pkg/cstorvolumereplica/v1alpha1/kubernetes.go

```diff
@@ -15,95 +15,164 @@
 	clientset "github.com/openebs/maya/pkg/client/generated/clientset/versioned"
-	kclient "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+
+	client "github.com/openebs/maya/pkg/kubernetes/client/v1alpha1"
 )
 
 // getClientsetFn is a typed function that
 // abstracts fetching of internal clientset
 type getClientsetFn func() (clientset *clientset.Clientset, err error)
 
+// getClientsetFromPathFn is a typed function that
+// abstracts fetching of clientset from kubeConfigPath
+type getClientsetForPathFn func(kubeConfigPath string) (clientset *clientset.Clientset, err error)
+
+// getFn is a typed function that abstracts get of cstorvolume replica instances
+type getFn func(cli *clientset.Clientset, name, namespace string,
+	opts metav1.GetOptions) (*apis.CStorVolumeReplica, error)
+
 // listFn is a typed function that abstracts
 // listing of cstor volume replica instances
 type listFn func(cli *clientset.Clientset, namespace string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error)
 
+// delFn is a typed function that abstracts delete of cstorvolume replica instances
+type delFn func(cli *clientset.Clientset, name, namespace string, opts *metav1.DeleteOptions) error
+
 // kubeclient enables kubernetes API operations
 // on cstor volume replica instance
-type kubeclient struct {
+type Kubeclient struct {
 	// clientset refers to cstor volume replica's
 	// clientset that will be responsible to
 	// make kubernetes API calls
 	clientset *clientset.Clientset
 
+	kubeConfigPath string
 	// namespace holds the namespace on which
 	// kubeclient has to operate
 	namespace string
 
 	// functions useful during mocking
-	getClientset getClientsetFn
-	list         listFn
+	getClientset        getClientsetFn
+	getClientsetForPath getClientsetForPathFn
+	get                 getFn
+	list                listFn
+	del                 delFn
 }
 
 // kubeclientBuildOption defines the abstraction
 // to build a kubeclient instance
-type kubeclientBuildOption func(*kubeclient)
+type kubeclientBuildOption func(*Kubeclient)
 
 // withDefaults sets the default options
 // of kubeclient instance
-func (k *kubeclient) withDefaults() {
+func (k *Kubeclient) withDefaults() {
+
 	if k.getClientset == nil {
 		k.getClientset = func() (clients *clientset.Clientset, err error) {
-			config, err := kclient.Config().Get()
+			config, err := client.GetConfig(client.New())
 			if err != nil {
 				return nil, err
 			}
 			return clientset.NewForConfig(config)
 		}
 	}
+
+	if k.list == nil {
+		k.list = func(cli *clientset.Clientset, namespace string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error) {
+			return cli.OpenebsV1alpha1().CStorVolumeReplicas(namespace).List(opts)
+		}
+	}
+
+	if k.getClientsetForPath == nil {
+		k.getClientsetForPath = func(kubeConfigPath string) (clients *clientset.Clientset, err error) {
+			config, err := client.GetConfig(client.New(client.WithKubeConfigPath(kubeConfigPath)))
```

Comment:
Please make use of `client.New(client.WithKubeConfigPath(...)).Config()` similar to other callers.